BedrockServer.h

```diff
@@ -277,8 +285,15 @@ class BedrockServer : public SQLiteServer {
     BedrockTimeoutCommandQueue _syncNodeQueuedCommands;
 
     // These control whether or not the command port is currently opened.
-    bool _suppressCommandPort;
-    bool _suppressCommandPortManualOverride;
+    multiset<string> _commandPortBlockReasons;
+
+    // This indicates if `_commandPortBlockReasons` is empty, *mostly*. It exists so that it can be checked without
+    // having to lock, like we do for _commandPortBlockReasons, and so it's feasible it gets out of sync for a few ms
+    // as we add and remove things from _commandPortBlockReasons, but it is only used as an optimization - we will
+    // close connections after commands complete if this is true, and not if it is false, so the worst-case scenario of
+    // this being wrong is we accept an extra command as the port is blocked, or cause a client to reconnect after a
+    // command is it's unblocked.
+    atomic<bool> _commandPortLikelyBlocked;
```

Comment:
How much performance gain do we get from this optimization? What is the cost of doing this vs locking on `_commandPortBlockReasons` to check its length?

Reply:
We don't lock `_commandPortBlockReasons`, we lock `_portMutex`. We also lock `_portMutex` for the main body of `_acceptSockets` and `_postPoll`, meaning we already don't do the same critical loop in those two functions simultaneously. Adding a lock on `_portMutex` here would mean we'd also potentially keep each `handleSocket` instance from running simultaneously (for one part of it's loop).

We can't really give actual performance numbers without benchmarking in production anyway, but I would really rather avoid locking any single resource inside `handleSocket` besides where we commit DB transactions.

Reply:
That makes sense. Might be helpful to add some of that color to the comment but NAB

Reply:
`_isCommandPortLikelyBlocked;`